keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/verification/VerificationRepository.kt

```diff
@@ -27,6 +27,14 @@ interface VerificationRepository {
   )
 
   fun nextEnvironmentsForVerification(minTimeSinceLastCheck: Duration, limit: Int) : Collection<VerificationContext>
+
+  fun wasSuccessfullyVerifiedIn(
+    deliveryConfig: DeliveryConfig,
+    artifactReference: String,
+    version: String,
+    environmentName: String,
```

Comment:
Consider passing a `DeliveryArtifact` and an `Environment` objects for parameters instead to avoid unnecessary changes to the signature later if we change the identifiers. Particularly for the artifact, where the reference alone as an identifier has been cause for debate (it also helps by allowing access to more details when logging).

Reply:
we've already got a `VerificationContext` object that's designed to abstract the details of these parameters that basically any method on this repo is going to need.

Reply:
Right, but that still has just the environment name and artifact reference, so if you wanted to access other properties you'd need to query for them (which we do all over our repositories and is wasteful).

Reply:
that's an implementation detail of that context class that we could change outside of the context of these methods